use_sim_time: false

env_type: 1
robot_type: 1         # 0: A1
                      # 1: Go1
controller_type: 1    # 0: QP
                      # 1: Convex MPC
                      # 2: Quaternion MPC
kf_type: 1            # 0: no kf ( only available in Gazebo)
                      # 1: linear KF
                      # 2: EKF
terrain_adpt_state: 0 # 0: turn off terrain adaptation
                      # 1: turn on terrain adaptation

# Default foothold location in Relative frame
default_foot_pos_FL_x: 0.2104
default_foot_pos_FL_y: 0.13
default_foot_pos_FL_z: -0.3

default_foot_pos_FR_x: 0.2104
default_foot_pos_FR_y: -0.13
default_foot_pos_FR_z: -0.3

default_foot_pos_RL_x: -0.1658
default_foot_pos_RL_y: 0.13
default_foot_pos_RL_z: -0.3

default_foot_pos_RR_x: -0.1658
default_foot_pos_RR_y: -0.13
default_foot_pos_RR_z: -0.3

# Gait frequency
gait_freq: 2.0 # gait cycle per second

# Convex MPC parameters
mpc_update_period: 8.0
mpc_horizon: 30

q_weights_0: 15.0 # roll
q_weights_1: 70.0 # pitch
q_weights_2: 10.0 # yaw

q_weights_3: 5.0 # pos x
q_weights_4: 5.0 # pos y
q_weights_5: 150.0 # pos z

q_weights_6: 0.02 # omega x
q_weights_7: 0.02 # omega y
q_weights_8: 10.0 # omega z

q_weights_9: 2.5 # vel x
q_weights_10: 2.5 # vel y
q_weights_11: 20.0 # vel z

r_weights_0: 0.0001
r_weights_1: 0.0001
r_weights_2: 0.0001

r_weights_3: 0.0001
r_weights_4: 0.0001
r_weights_5: 0.0001

r_weights_6: 0.0001
r_weights_7: 0.0001
r_weights_8: 0.0001

r_weights_9: 0.0001
r_weights_10: 0.0001
r_weights_11: 0.0001

mu: 0.7
fz_max: 500.0

# Swing leg control parameters
kp_hip: 25.0
kp_thigh: 25.0
kp_calf: 25.0

kd_hip: 2.0
kd_thigh: 2.0
kd_calf: 2.0

# Joystick settings
joystick_left_updown_axis:      1
joystick_left_horiz_axis:       0
joystick_right_updown_axis:     4
joystick_right_horiz_axis:      3
joystick_cross_left_right_axis: 6
joystick_cross_up_down_axis:    7

joystick_mode_switch_button:    0 # A
joystick_default_pos_button:    1 # B
joystick_zero_torque_button:    2 # X
sin_ang_vel_button:             3 # Y
joystick_exit_button:           4 # LB

# Joystick scaling
joystick_velx_scale: 0.3  # m/s
joystick_vely_scale: 0.3   # m/s
joystick_height_vel: 0.1   # m/s
joystick_max_height: 0.30  # m
joystick_min_height: 0.03  # m
joystick_yaw_rate_scale: 1.57  # rad/s
joystick_roll_rate_scale: 0.0  # rad/s
joystick_pitch_rate_scale: 0.0  # rad/s

# Contact detection related 
foot_sensor_max_value: 150.0
foot_sensor_min_value: 0.0
foot_sensor_ratio: 0.5

# casadi EKF parameters
ekf_inital_cov:               0.01
ekf_noise_process_pos_xy:     0.01
ekf_noise_process_pos_z:      0.01
ekf_noise_process_vel_xy:     0.5
ekf_noise_process_vel_z:      0.5
ekf_noise_process_rot:        0.00001
ekf_noise_process_foot:       0.01

ekf_noise_measure_fk:         0.01
ekf_noise_measure_vel:        0.5
ekf_noise_measure_height:     99.0

ekf_noise_opti_pos:           0.001
ekf_noise_opti_vel:           999.0
ekf_noise_opti_yaw:           0.001

# Robot parameters
robot_mass: 12.0

trunk_inertia_xx: 0.0168128557
trunk_inertia_xy: 0.0
trunk_inertia_xz: 0.0
trunk_inertia_yz: 0.0
trunk_inertia_yy: 0.063009565
trunk_inertia_zz: 0.0716547275
